### PR TITLE
Helm & the gRPC Experience

### PR DESCRIPTION
**Pull request by fibonacci1729**
@zinuga

### CODE REVIEW

**Comment by zinuga** [comment]
@LisaFC can you review and publish

File: _posts/helm-grpc.md [hunk 1]
```diff
@@ -0,0 +1,43 @@
+---
+layout: post
+title:  gRPC in Helm
+---
+
+Helm is the package manager for Kubernetes. Helm provides its users with a customizable mechanism for
+managing distributed applications and controlling their deployment. 
+
+I have the fortune of being a member of the phenomenal open-source Kubernetes Helm community serving as
```
Comment (LisaFC) [inline _posts/helm-grpc.md]: I have the fortune of being -> I have the good fortune to be

File: _posts/helm-grpc.md [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+title:  gRPC in Helm
+---
+
+Helm is the package manager for Kubernetes. Helm provides its users with a customizable mechanism for
+managing distributed applications and controlling their deployment. 
+
+I have the fortune of being a member of the phenomenal open-source Kubernetes Helm community serving as 
+a core contributor. My first day working with the Helm team was spent prototyping the architecture for
+the next generation of Helm. By the end of that day, we had procured the preliminary RPC protocol data model
+used to enable communication between Helm and it's in-cluster server component, Tiller.
```
Comment (LisaFC) [inline _posts/helm-grpc.md]: it's -> its

File: _posts/helm-grpc.md [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+
+We chose to use protocol buffers as our data definition language (the default framework gRPC uses for
```
Comment (LisaFC) [inline _posts/helm-grpc.md]: We chose to use protocol buffers - the default framework gRPC uses for serialization and over-the-air transmission - as our data definition language.

File: _posts/helm-grpc.md [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+serialization and over-the-air transmission.) By the end of that first day hacking with the Helm team,
+gRPC and protocol buffers proved to be a powerful combination. We successfully had acheived communication
```
Comment (LisaFC) [inline _posts/helm-grpc.md]: successfully acheived -> had successfully achieved

File: _posts/helm-grpc.md [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+between the Helm client and Tiller server using code generated from the protobuf and gRPC service definitions.
+As a personal preference, relative to something like swagger, the protobuf files and resulting generated gRPC
```
Comment (LisaFC) [inline _posts/helm-grpc.md]: As a personal preference, we found that the protobuf files and resulting generated gRPC code provided an aesthetic, nearly self-documenting developer experience compared to something like Swagger.

File: _posts/helm-grpc.md [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+code provided an aesthetic, nearly self-documenting developer experience.
+
+Within a few days, the helm team was scoping and implementing features for our users. By choosing gRPC/Proto
```
Comment (LisaFC) [inline _posts/helm-grpc.md]: the helm team -> the Helm team

File: _posts/helm-grpc.md [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+we had reduced the typical time spent bikeshedding that, in general, inevitably evolves from API modeling and
+churning out boilerplate server code. If we had not reaped the benefits of gRPC/protobuf from day 1, we would
+have spent significantly more time pivoting up and down the stack, as opposed to honing our focus on what
+matters: the users and the features they requested.
+
+In addition to serving as the Helm/Tiller communication protocol, one of our more interesting applications
+of protocol buffers is that we use it to model what's referred to in Kubernetes parlance, as a "chart". Charts
```
Comment (LisaFC) [inline _posts/helm-grpc.md]: what's referred to in Kubernetes parlance as a "chart".

File: _posts/helm-grpc.md [hunk 8]
```diff
@@ -0,0 +1,43 @@
...
+are an encapsulation of kubernetes manifests that enable you to define, install, and upgrade Kubernetes applications.
```
Comment (LisaFC) [inline _posts/helm-grpc.md]: Kubernetes manifests

File: _posts/helm-grpc.md [hunk 9]
```diff
@@ -0,0 +1,43 @@
...
+For the more complex Kubernetes applications, the set of manifests may be large. By virtue of its inherent compression
```
Comment (LisaFC) [inline _posts/helm-grpc.md]: For more complex Kubernetes applications,...

File: _posts/helm-grpc.md [hunk 10]
```diff
@@ -0,0 +1,43 @@
...
+capabilities, protocol buffer and gRPC allowed us to mitigate the nuisance of transmitting bulky and
```
Comment (LisaFC) [inline _posts/helm-grpc.md]: protocol buffers and gRPC

File: _posts/helm-grpc.md [hunk 11]
```diff
@@ -0,0 +1,43 @@
...
+sprawling Kubernets manifests.
```
Comment (LisaFC) [inline _posts/helm-grpc.md]: Kubernets -> Kubernetes

File: _posts/helm-grpc.md [hunk 12]
```diff
@@ -0,0 +1,43 @@
...
+
+For a deeper dive into,
```
Comment (LisaFC) [inline _posts/helm-grpc.md]: into, -> into:

File: _posts/helm-grpc.md [hunk 13]
```diff
@@ -0,0 +1,43 @@
...
+	- The Helm proto, see: https://github.com/kubernetes/helm/_proto/hapi
+	- Its generated counterpart, see: https://github.com/kubernetes/helm/pkg/proto/hapi
+	- The interface to our helm client, see: https://github.com/kubernetes/helm/pkg/helm
```
Comment (LisaFC) [inline _posts/helm-grpc.md]: helm client -> Helm client

**Comment by fibonacci1729** [comment]
@LisaFC Thank you for the feedback. I believe I addressed all the corrections.

**Pull request merged**